metricproviders/wavefront/wavefront.go

```diff
@@ -37,55 +37,67 @@ func (p *Provider) Type() string {
 }
 
 type WavefrontClientAPI interface {
-	NewQuery(params *wavefront_api.QueryParams) WavefrontQueryAPI
+	NewQuery(params *wavefrontapi.QueryParams) WavefrontQueryAPI
 }
 
 type WavefrontClient struct {
-	*wavefront_api.Client
+	*wavefrontapi.Client
 }
 
-func (wc *WavefrontClient) NewQuery(params *wavefront_api.QueryParams) WavefrontQueryAPI {
+func (wc *WavefrontClient) NewQuery(params *wavefrontapi.QueryParams) WavefrontQueryAPI {
 	return &WavefrontQuery{Query: wc.Client.NewQuery(params)}
 }
 
 type WavefrontQueryAPI interface {
-	Execute() (*wavefront_api.QueryResponse, error)
+	Execute() (*wavefrontapi.QueryResponse, error)
 }
 
 type WavefrontQuery struct {
-	*wavefront_api.Query
+	*wavefrontapi.Query
 }
 
-func (wq *WavefrontQuery) Execute() (*wavefront_api.QueryResponse, error) {
+func (wq *WavefrontQuery) Execute() (*wavefrontapi.QueryResponse, error) {
 	return wq.Query.Execute()
 }
 
+type wavefrontResponse struct {
+	newValue   string
+	newStatus  v1alpha1.AnalysisPhase
+	epochsUsed string
+}
+
 // Run queries with wavefront provider for the metric
 func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alpha1.Measurement {
 	startTime := metav1.Now()
 	newMeasurement := v1alpha1.Measurement{
 		StartedAt: &startTime,
+		Metadata:  map[string]string{},
 	}
 
-	queryParams := &wavefront_api.QueryParams{
+	queryParams := &wavefrontapi.QueryParams{
 		QueryString:             metric.Provider.Wavefront.Query,
 		StartTime:               strconv.FormatInt(time.Now().Unix()*1000, 10),
+		EndTime:                 strconv.FormatInt(time.Now().Unix()*1000, 10),
 		MaxPoints:               "1",
 		Granularity:             "s",
 		SeriesOutsideTimeWindow: false,
 	}
 
 	response, err := p.api.NewQuery(queryParams).Execute()
+	if response != nil && response.Warnings != "" {
+		newMeasurement.Metadata["warnings"] = response.Warnings
+	}
 	if err != nil {
 		return metricutil.MarkMeasurementError(newMeasurement, err)
 	}
-	newValue, newStatus, err := p.processResponse(metric, response)
+	result, err := p.processResponse(metric, response, startTime)
 	if err != nil {
 		return metricutil.MarkMeasurementError(newMeasurement, err)
 
 	}
-	newMeasurement.Value = newValue
-	newMeasurement.Phase = newStatus
+	newMeasurement.Value = result.newValue
+	newMeasurement.Phase = result.newStatus
+	newMeasurement.Metadata["epochs-used"] = result.epochsUsed
```

Comment:
1. "epochs" is the plural of epoch, which in this case doesn't make sense, since it is a single value
2. i think we can just call this `time`, which is the terminology prometheus and wavefront seem to use.

Reply:
We discussed offline and decided on timestamps since there can be multiple timestamps if there are multiple timeseries